aiohttp/helpers.py

```diff
@@ -831,9 +831,15 @@ def __repr__(self) -> str:
 
 
 class CookieMixin:
+    # The `_cookies` slots is not defined here because non-empty slots cannot
+    # be combined with an Exception base class, as is done in HTTPException.
+    # CookieMixin subclasses with slots should
+    # define the `_cookies` slot themselves.
+    __slots__ = ()
+
     def __init__(self) -> None:
         super().__init__()
-        self._cookies: SimpleCookie[str] = SimpleCookie()
+        self._cookies: SimpleCookie[str] = SimpleCookie()  # type: ignore
```

Comment:
I don't know either what this 'lay-out' conflict is, exactly. It may have to do with the fact that `Exception` is a builtin class, but its instances _do_ have a `__dict__`.

Reply:
From the documentation there are some _hints_:

> Multiple inheritance with multiple slotted parent classes can be used, but only one parent is allowed to have attributes created by slots (the other bases must have empty slot layouts) - violations raise TypeError.

Reply:
The error is supposed to indicate an issue where it's unclear which class the attribute would come from:
```
>>> class A:            
...   __slots__ = ("a",)
... 
>>> class B:            
...   __slots__ = ("a",)
... 
>>> class C(A, B):
...  __slots__ = ()
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: multiple bases have instance lay-out conflict
```

It's interesting that the same error occurs with `int`. This doesn't support non-empty slots due to being variable length, which would normally give an error like:
```
>>> class C(int):
...  __slots__ = ("c",)
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: nonempty __slots__ not supported for subtype of 'int'
```

Maybe there is some other restriction with C classes that I couldn't find documented.

Reply:
I've added an extra explanation on the mypy ignore. I'll consider it resolved unless anybody else comments :)